compiler/rustc_hir_typeck/src/closure.rs

```diff
@@ -724,20 +726,17 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 .find_map(|(p, s)| get_future_output(p.as_predicate(), s))?,
             ty::Error(_) => return None,
             _ => span_bug!(
-                self.tcx.def_span(expr_def_id),
+                span,
                 "async fn generator return type not an inference variable: {ret_ty}"
             ),
         };
 
+        let output_ty = self.normalize(span, output_ty);
```

Comment:
can you move the `normalize` call into the `ty::Alias` branch? for infer vars the obligation should already be normalized

Reply:
That's the alias that corresponds to the `impl Future`. The type we want to normalize here is the `Output` of that future, which depends on what we extract out of `get_future_output`.

Reply:
Wait, wait, I think I understand here.

Reply:
Actually, we still may have unnormalized obligations for infer vars as well. The obligations returned from `replace_opaque_with_infer` and `handle_opaque_ty` are not normalized, afaict.